# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1346514612 14400
#      Sat Sep 01 11:50:12 2012 -0400
# Node ID 535bb601e7db880c79edfb7bc8650538fd8edf37
# Parent  dd794aa84232439db94a48fe22038cc4ba078a7b
use absolute file name to find mk-ops.awk

* liboctave/operators/module.mk (OP_SRCDIR): Use absolute name.
Let make find operator file dependencies with VPATH.

diff --git a/liboctave/operators/module.mk b/liboctave/operators/module.mk
--- a/liboctave/operators/module.mk
+++ b/liboctave/operators/module.mk
@@ -37,29 +37,29 @@ OPERATORS_INC = \
   operators/Sparse-perm-op-defs.h
 
 ## There are no distributed source files in this directory
 OPERATORS_SRC =
 
 OTHER_INC += \
   operators/mx-inlines.cc
 
-OP_SRCDIR = $(srcdir)/operators
+OP_SRCDIR = $(abs_top_srcdir)/liboctave/operators
 
 ## Special rules for sources which must be built before rest of compilation.
-$(VX_OP_INC) $(VX_OP_SRC) : $(OP_SRCDIR)/mk-ops.awk operators/vx-ops
+$(VX_OP_INC) $(VX_OP_SRC) : operators/mk-ops.awk operators/vx-ops
 	(cd operators; $(AWK) -f $(OP_SRCDIR)/mk-ops.awk prefix=vx $(OP_SRCDIR)/vx-ops)
 
-$(MX_OP_INC) $(MX_OP_SRC) : $(OP_SRCDIR)/mk-ops.awk operators/mx-ops
+$(MX_OP_INC) $(MX_OP_SRC) : operators/mk-ops.awk operators/mx-ops
 	(cd operators; $(AWK) -f $(OP_SRCDIR)/mk-ops.awk prefix=mx $(OP_SRCDIR)/mx-ops)
 
-$(SMX_OP_INC) $(SMX_OP_SRC) : $(OP_SRCDIR)/sparse-mk-ops.awk operators/sparse-mx-ops
+$(SMX_OP_INC) $(SMX_OP_SRC) : operators/sparse-mk-ops.awk operators/sparse-mx-ops
 	(cd operators; $(AWK) -f $(OP_SRCDIR)/sparse-mk-ops.awk prefix=smx $(OP_SRCDIR)/sparse-mx-ops)
 
-operators/mx-ops.h : $(OP_SRCDIR)/mk-ops.awk operators/mx-ops
+operators/mx-ops.h : operators/mk-ops.awk operators/mx-ops
 	$(AWK) -f $(OP_SRCDIR)/mk-ops.awk prefix=mx make_inclusive_header=mx-ops.h $(OP_SRCDIR)/mx-ops > $@-t
 	mv $@-t $@
 
 noinst_LTLIBRARIES += operators/liboperators.la
 
 operators_liboperators_la_SOURCES = $(OPERATORS_SRC)
 nodist_operators_liboperators_la_SOURCES = $(BUILT_LIBOPERATORS_SOURCES)
 
